### PR TITLE
Factor out global configuration into a dedicated class

### PR DESCRIPTION
**Pull request by CyberDem0n**
It will help to avoid code duplications.

### CODE REVIEW

File: patroni/config.py [hunk 1]
```diff
@@ -35,6 +37,72 @@ def default_validator(conf):
...
+        """:returns: "standby_cluster" configuration."""
+        return self.get('standby_cluster')
+
+    def is_standby_cluster(self) -> bool:
+        """:returns: `True` if global configuration has a valid "standby_cluster" section."""
+        config = self.get_standby_cluster_config()
+        return isinstance(config, dict) and (config.get('host') or config.get('port') or config.get('restore_command'))
+
+
+def get_global_config(cluster: Union[Cluster, None], default: Optional[Dict] = None):
```
Comment (hughcapet) [inline patroni/config.py:91]: return value

File: patroni/utils.py [hunk 2]
```diff
@@ -693,19 +694,7 @@ def iter_response_objects(response: HTTPResponse) -> Iterator[Dict[str, Any]]:
...
-    :returns: ``True`` if configuration is a Patroni standby cluster.
-    """
-    return isinstance(config, dict) and (config.get('host') or config.get('port') or config.get('restore_command'))
-
-
-def cluster_as_json(cluster: 'Cluster') -> Dict[str, Any]:
+def cluster_as_json(cluster: 'Cluster', global_config: Optional['GlobalConfig'] = None) -> Dict[str, Any]:
     """Get a JSON representation of *cluster*.
 
     :param cluster: the :class:`Cluster` object to be parsed as JSON.
```
Comment (hughcapet) [inline patroni/utils.py:700]: :param global_config:

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**